artifactory/services/utils/properties.go

```diff
@@ -35,33 +36,56 @@ func ParseProperties(propStr string) (*Properties, error) {
 }
 
 func (props *Properties) ParseAndAddProperties(propStr string) error {
-	propList := strings.Split(propStr, propsSeparator)
+	propList := splitWhileIgnoringBackslashPrefixSeparators(propStr, propsSeparator)
 	for _, prop := range propList {
 		if prop == "" {
 			continue
 		}
 
-		parts := strings.Split(prop, "=")
-		if len(parts) != 2 {
-			return errorutils.CheckError(errors.New(fmt.Sprintf("Invalid property format: %s - format should be key=val1,val2,...", prop)))
+		key, value, err := splitPropToKeyAndValue(prop)
+		if err != nil {
+			return err
 		}
 
-		key := parts[0]
-		values := strings.Split(parts[1], valuesSeparator)
-		for i, val := range values {
-			// If "\" is found, then it means that the original string contains the "\," which indicates this "," is part of the value
-			// and not a separator
-			if strings.HasSuffix(val, "\\") && i+1 < len(values) {
-				values[i+1] = val[:len(val)-1] + valuesSeparator + values[i+1]
-			} else {
-				props.properties[key] = append(props.properties[key], val)
-			}
+		splitValues := splitWhileIgnoringBackslashPrefixSeparators(value, valuesSeparator)
+		for _, val := range splitValues {
+			props.properties[key] = append(props.properties[key], val)
 		}
 	}
 	props.removeDuplicateValues()
 	return nil
 }
 
+// Searches for the first "=" instance, and split str into 2 substrings.
+// Returns error for invalid property format.
+func splitPropToKeyAndValue(str string) (key, value string, err error) {
+	index := strings.Index(str, keyValuesSeparator)
+	if index == -1 || index-1 < 0 || index+1 >= len(str) {
+		return "", "", errorutils.CheckError(errors.New(fmt.Sprintf("Invalid property format: %s - format should be key=val1,val2,...", str)))
+	}
+	key = str[0:index]
+	value = str[index+1:]
+	err = nil
+	return
+}
+
+// Split slices s into all substrings separated by sep and returns a slice of the substrings between those separators,
+// ignoring separators with a '\' prefix, which indicates that the separator is a part of the value.
```

Comment:
// Split slices s into all substrings separated by sep and returns a slice of the substrings between those separators,
// ignoring separators with a '\' prefix, which indicates that the separator is a part of the value.
-->
// Returns a slice created by splitting the sent string s into substrings, using the sent separator.
// A backslash prefix escapes the separator.